src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-node-statistics.ts

```diff
@@ -0,0 +1,478 @@
+    }
+    return rssi.toString();
+  }
+
+  private _computeDeviceNameById(device_id: string): "unknown device" | string {
+    if (!this._deviceIDsToDevices) {
+      return "unknown device";
+    }
+    const device = this._deviceIDsToDevices[device_id];
+    if (!device) {
+      return "unknown device";
+    }
+
+    return computeDeviceName(device, this.hass);
+  }
+
+  private _subscribeNodeStatistics(): void {
+    if (!this.hass) {
+      return;
+    }
+    this._subscribedNodeStatistics = subscribeZwaveNodeStatistics(
+      this.hass,
+      this.device!.id,
+      (message: ZWaveJSNodeStatisticsUpdatedMessage) => {
+        this._nodeStatistics = message;
+
+        if (this._nodeStatistics.rssi) {
+          this._nodeStatistics.rssi_translated = this._computeRSSI(
+            this._nodeStatistics.rssi,
+            false
+          );
+        }
+
+        const workingRoutes: [
+          string,
+          WorkingRouteStatistics | null | undefined
+        ][] = [
+          ["lwr", this._nodeStatistics?.lwr],
+          ["nlwr", this._nodeStatistics?.nlwr],
+        ];
+
+        this._workingRoutes = {};
+        workingRoutes.forEach(([wrKey, wrValue]) => {
+          this._workingRoutes[wrKey] = wrValue;
+
+          if (wrValue) {
+            if (wrValue.rssi) {
+              wrValue.rssi_translated = this._computeRSSI(wrValue.rssi, true);
+            }
+
+            if (wrValue.route_failed_between) {
+              wrValue.route_failed_between_translated = [
+                this._computeDeviceNameById(wrValue.route_failed_between[0]),
+                this._computeDeviceNameById(wrValue.route_failed_between[1]),
+              ];
+            }
+
+            if (wrValue.repeaters && wrValue.repeaters.length) {
+              wrValue.repeater_rssi_table = html`${wrValue.repeaters.map(
+                (_, idx) =>
+                  html`<div class="row">
+                    <span class="key-cell"
+                      >${this._computeDeviceNameById(
+                        wrValue.repeaters[idx]
+                      )}:</span
+                    >
+                    <span class="value-cell"
+                      >${this._computeRSSI(
+                        wrValue.repeater_rssi[idx],
+                        true
+                      )}</span
+                    >
+                  </div>`
+              )}`;
+            }
+          }
+        });
+      }
+    );
+  }
+
+  private _subscribeDeviceRegistry(): void {
+    if (!this.hass) {
+      return;
+    }
+    this._subscribedDeviceRegistry = subscribeDeviceRegistry(
+      this.hass.connection,
+      (devices: DeviceRegistryEntry[]) => {
+        Object.entries(this._deviceIDsToDevices).forEach(([_, device]) => {
+          if (!devices.includes(device))
+            delete this._deviceIDsToDevices[device.id];
+        });
+        devices.forEach((device) => {
+          this._deviceIDsToDevices[device.id] = device;
+        });
```

Comment:
```suggestion
        const devicesIdToName = {};
        devices.forEach(device => {devicesIdToName[device.id] = computeDeviceName(device, this.hass)});
        this._deviceIDsToName = devicesIdToName;
```

This makes sure it triggers an update when the devices are updated.

Reply:
I applied the same approach to `this._workingRoutes`